### ISSUE TITLE
Don't publish empty data

### ISSUE THREAD

**Issue by tkurki** [htool/vhfinfo#14]
A large portion of my Data Browser looks like this:
<img width="427" alt="image" src="https://github.com/htool/vhfinfo/assets/1049678/37e8ac6d-1937-449f-9a87-a6169132a7f5">

**Comment by htool** [comment]
To allow displaying the channel number and other info I'm sending the sorted info to SignalK paths.
When the next list is shorter, as far as I know there's no way to delete an old entry.
So as a workaround, I'm overwriting with blank.
What would be a better way to do this?

**Comment by tkurki** [comment]
It sounds like what is happening is that you are _publishing a new list_. For that operation / use case it would be much simpler if the whole list would be a single value, an array, for a single path `vhfdata.nearest`.

**Comment by tkurki** [comment]
I don't see how that would actually work. Here a certain "path" is arbitrary - now the channel I am interested is 3rd on the list, but as I maneuver in a moment it may be 2nd or 4th or something else.

I think the static part of this data should be modeled as resources, retrievable via the resources api, and the updates/delta data would be just the relative data: resource id, distance & bearing (ouch I just noticed `bearing` and `bearingRad` - everything in SK should use radians!).

I can help with the implementation.

**Comment by htool** [comment]
Agreed on the resource model approach. That was the plan all along, but to first get something up and running to understand better the use cases. Quite some things changed since the start around assumptions.
E.g. I started with a channel numeric. Then learned there are countries using '5a'. And later found that certain stations in Denmark have 3 channels. 16, 13 or 12.

Now that the format is at least more stable it can move to a proper place. I haven't done any resource provider yet.

Currently time is somewhat limited, but after the sail season I can put time in again to get this migrated over the winter period all ready in time for next season.

**Comment by htool** [comment]
Proposal: VHF coverage as a custom Resource (vhfinfo)

Register app.registerResourceProvider({ type: 'vhfinfo' }) and serve GeoJSON polygons through the standard Resources API, using the Region record shape (name, description, feature.geometry Polygon/MultiPolygon, extra fields in feature.properties).

Why a custom type, not regions: VHF cells are regions in the geometric sense, but they must not land in GET /signalk/v2/api/resources/regions. Without ?provider=, Signal K merges every regions provider. Chart UIs that list regions would then paint every VTS, lock, and coast-radio polygon. A separate type keeps charts clean; clients that already understand Region can still parse the same JSON. POST/PUT/DELETE stay unimplemented — editing remains on vhfinfo.org.

Where distance lives: Resource list queries use the OpenAPI spatial filters only: bbox, distance (square in metres around the vessel or position), position, limit. That is the catalog (“what is in this box?”). Ranked nearby (search beam, signed boundary distance, information-only-when-inside, sort by centroid) is computed state, not a Region query. It stays on GET /signalk/v1/api/vhfinfo/nearby (readonly) and can be mirrored later. Per-feature distance/bearing belong on that nearby payload (and optionally one compact delta), not as exploded vessels.self paths.

How v2 is used:

Call	Role
`GET /signalk/v2/api/resources/vhfinfo?bbox=…`
Catalog in view
`GET /signalk/v2/api/resources/vhfinfo?distance=4000`
Catalog around the boat
`GET /signalk/v2/api/resources/vhfinfo/{id}`
One coverage polygon
`GET /signalk/v1/api/vhfinfo/nearby`
Ranked live list (compat + MFD)

Stop writing leaf paths under vhfdata.nearest.*. Keep the JSON blobs (and/or one compact vhfinfo.nearby array) until MFD and instrument-display read Resources + /nearby.

@tkurki Do you agree with the above approach?

**Comment by tkurki** [comment]
I'd prefer the opposite, using the argument in reverse: use `regions`, not a custom type, exactly so that clients that already support them will show them. If the result is a mess it pushes us to refine the resources api, to create either more custom resource types or add metadata so that users can further specify how and what they want displayed.

Having a custom resource type means that all clients must change, but it should be easy enough to add a configuration switch for regions:
- ( ) no resource API integration
- ( ) provide `region` resources
- ( ) provide `vhfinfo` resources
